.github/buildomat/jobs/image.sh

```diff
@@ -28,12 +28,16 @@ rustc --version
 
 banner build
 
-# Enable the "omicron-build" feature to indicate this is an artifact destined
-# for production use on an appropriately configured Oxide machine
+# This job produces the propolis-server zone image that the Omicron repository
+# imports to create full-blown Oxide software packages. Disable all default
+# features and opt into the "omicron-build" feature instead to build libraries
+# in their "production" configurations.
 #
 # The 'release' profile is configured for abort-on-panic, so we get an
 # immediate coredump rather than unwinding in the case of an error.
-ptime -m cargo build --release --verbose -p propolis-server --features omicron-build
+ptime -m cargo build --release --verbose -p propolis-server \
+    --no-default-features \
+    --features omicron-build
```

Comment:
it took me a second to connect the dots but from #755 and elsewhere we're clearly still collecting cores from a crashed Propolis - all we've done by disabling the logging is prevented ourselves from collecting more timely information (that in the case of #755 helped us tell the guest had gone off to space)

i think a consistent posture here would be either:

A: Propolis stderr/stdout is as sensitive as guest memory; do not collect, if you must collect then keep secret, if you must share, then do so carefully
B: keep Propolis stderr/stdout clean of user data, and when operating Propolis, do not retain user data (cores that could include pages of user data)

i'm inclined towards A on the grounds that if you're looking at or copying a Propolis log around, you probably also can access a core from that same process. and it means we're actually helping a situation like #755 with this change.

i still think the _feature_ is useful, just that we can leave default features enabled in this job. _or_ if we should consider ourselves responsible for keeping Propolis lib/standalone/server's logs free of potentially-sensitive bits, it'd be worth a note to ourselves that we should audit other logs for anything untoward - migration state comes to mind.

i don't want to be excessively black/white on this, but i'm trying to make sure there's a clear line for what is and isn't OK for Propolis to say. what do you think?

Reply:
I keep finding myself drawn to approach B. At some point we're going to have a situation where we need to debug a Propolis issue using only information from operator-provided support bundles. That's going to be tricky if operators feel they can't ever include Propolis logs in those bundles because Propolis is too cavalier about logging guest data.

> i'm inclined towards A on the grounds that if you're looking at or copying a Propolis log around, you probably also can access a core from that same process.

I would put it a bit differently: if you can *extract a Propolis log from a sled*, you can probably also extract core dumps from that sled. Because Oxide personnel are regularly in this position today (because we access deployed racks over the technician port in order to update/service them), I agree that redacting diagnostic data in logs currently has very little practical upside, at least from the point of view of an operator who wants to limit support personnel access to guest VM state. The long term goal, of course, is to wean ourselves off of the tech port via automated update, support bundles, and the like, and once we've done that I think redaction will be more obviously useful.

#755 is the only case I know of where we've seen one of the diagnostic-generating faults in a production Omicron deployment. All the other cases I know of have been in development environments (ad hoc Propolis server/standalone binaries or `a4x2` deployments). So my lean was (and still is, I think) toward disabling this by default in Omicron images, though I think it's a somewhat close call (made even closer by the fact that #755 happened on an Oxide-managed rack and not a customer-operated rack).

With all that said: WWYT of taking some steps now to make this into a runtime-configurable setting? I think that the ideal world is one @pfmooney laid out in chat in yesterday's HV huddle: if this behavior can change dynamically, and Nexus knows how to control it, then an operator who can reproduce a problem in a "clean" VM can enable extra data collection for just that instance (or its project/silo/what-have-you) and provide us with the resulting "full-fat" logs without worry. In the short run (i.e. before Omicron learns about the new API), and in a dire pinch, support personnel accessing a rack through the tech port can access the API, enabling a "user starts clean VM -> support person sets option -> user repros problem" approach if we desperately needed it. I think I can adjust this PR to move in that direction pretty easily (and I suspect it might improve its ergonomics in the process); will probably go ahead and give that a try, at least.

Reply:
> wean ourselves off of the tech port via automated update, support bundles, and the like, and once we've done that I think redaction will be more obviously useful.

ah, yeah, in that context i see what you mean. i was over-indexing on today's diagnostics!

definitely a fan of making it runtime-configurable, then, but i'm even more glad to have piped up on RFD 496 - seems like we need a more nuanced treatment of the relationship between cores and support bundles.